src/output/help.rs

```diff
@@ -1124,10 +1043,29 @@ impl<'help, 'app, 'parser, 'writer> Help<'help, 'app, 'parser, 'writer> {
                     self.write_subcommands(self.parser.app)?;
                 }
                 b"after-help" => {
-                    self.none(self.parser.app.after_help.unwrap_or("unknown after-help"))?;
+                    let after_help = if self.use_long {
+                        self.parser
+                            .app
+                            .after_long_help
+                            .or(self.parser.app.after_help)
+                    } else {
+                        self.parser.app.after_help
+                    };
+                    if let Some(output) = after_help {
+                        self.none("\n\n")?;
```

Comment:
Can you add a small test that generates help of a program which has after_help but no args at all? I think you will see that there will be 4 empty lines because [this](https://github.com/clap-rs/clap/pull/2067/files#diff-750408d27c2eebbe9c64eccbc80aac19L915-L917) condition is not used.

Reply:
Good eye! That scenario does indeed produce 4 empty lines, but only when the default args for `--help` and `--version` are also disabled.

I can fix it by [using a different template when there are no args](https://github.com/mkantor/clap/compare/default-template...WIP/default-template-different-no-arg-template).

Alternatively, I could:

1. [Check for args while expanding `{after-help}`, don't emit `\n\n` when there aren't any](https://github.com/mkantor/clap/compare/default-template...WIP/default-template-magic-padding-for-after-help).

   Downside: this bakes assumptions about the overall template into `{after-help}` (namely that it will always come right after `{all-args}`). That's the case for the default template, but might not be for user-defined templates.

1. Do nothing; leave the code as-is.

   If this configuration (no user-defined args, `--help` and `--version`  disabled, and `after_help` set to something) is highly unusual, maybe it's okay if it looks a little weird.

Let me know what you prefer.

Reply:
Using a different template sounds like the correct solution.

Reply:
Done: 85d3daa.